alphalens/performance.py

```diff
@@ -527,3 +529,109 @@ def average_cumulative_return(q_fact, demean_by):
             fq = factor_data['factor_quantile']
             return fq.groupby(fq).apply(average_cumulative_return, None)
+
+
+def calc_vifs(data):
+    '''
+    Computes variance inflation factors (VIFs) for a set of random variables.
+    For more information, see
+    https://en.wikipedia.org/wiki/Variance_inflation_factor
+    Parameters
+    ----------
+    data : pd.DataFrame
+        DataFrame with observations of random variables.
+        - Columns are observations of a random variable, and each row makes up
+        one observation for each random variable.
+    '''
+    vifs = np.zeros(len(data.columns))
+    for i in range(len(data.columns)):
+        r2 = (OLS(data.iloc[:, i],
+                  add_constant(data.drop(data.columns[i], axis='columns')))
+              .fit()
+              .rsquared)
+        vifs[i] = 1/(1-r2)
+    return pd.Series(index=data.columns, data=vifs)
+
+
+def decompose_returns(algo_returns, risk_factors, hierarchy=None):
+    '''
+    Decomposes returns into returns attributable to common risk factors.
+    Parameters
+    ----------
+    algo_returns : pd.Series
+        Algorithm's daily returns, with a tz-localized index
+        - Example:
+        2010-01-01 00:00:00+00:00    0.812385
+        2010-01-04 00:00:00+00:00    0.468386
+        2010-01-05 00:00:00+00:00    0.680892
+    risk_factors : pd.DataFrame
+        Daily common risk factors. Index is tz-localized DatetimeIndex,
+        columns are common factor names
+        - Example:
+                                        Mkt       SMB       HML       UMD
+        2017-06-26 00:00:00+00:00    0.0004   -0.0008    0.0069   -0.0057
+        2017-06-27 00:00:00+00:00   -0.0084   -0.0043    0.0130   -0.0030
+        2017-06-28 00:00:00+00:00    0.0102    0.0077    0.0021    0.0091
+    hierarchy : OrderedDict
+        Hierarchy of common factors, for use in plotting the bar chart.
+        If None, defaults to a hierarchy in which each column of risk_factors
+        forms a 'level' of the hierarchy.
+        - Example:
+        OrderedDict([('Market', ['Mkt']),
+                     ('Style', ['SMB', 'HML', 'UMD'])])
+    Returns
+    -------
+    returns_decomposition : pd.DataFrame
+        Decomposed returns
+    betas : pd.DataFrame
+        Betas to the common factors
+    t_stats : pd.DataFrame
+        t-statistics for the betas to the common factors. Computed by dividing
+        betas by White's (1980) heteroskedasticity standard errors.
+    vifs : pd.Series
+        Variance inflation factors for the common factors
+        - For more information, see
+        https://en.wikipedia.org/wiki/Variance_inflation_factor
+    '''
+    if hierarchy is None:
+        hierarchy = risk_factors.columns.values
+
+    # construct a cumulative hierarchy
+    cum_tiers = []
+    for i in range(len(hierarchy.values())):
+        cum_tiers.append([item for a in range(0, i+1)
+                          for item in hierarchy.values()[a]])
```

Comment:
Getting dictionary keys and values changed in Python 3. I think `hierarchy.values()` in line 603 should be replaced with `list(hierarchy.values())` when you try to access it with brackets.